src/ray/core_worker/transport/direct_task_transport.h

```diff
@@ -72,6 +72,11 @@ class CoreWorkerDirectTaskSubmitter {
   /// \param[in] task_spec The task to schedule.
   Status SubmitTask(TaskSpecification task_spec);
 
+  /// Either remove a pending task or send an RPC to kill a running task
+  ///
+  /// \param[in] task_spec The task to kill.
+  Status KillTask(TaskSpecification task_spec, bool force_kill, int num_tries=3);
```

Comment:
I wanted to keep it finite because there it is possible for the task to finish execution at a time after the cancellation has been sent and before it has been processed. It would not make sense to keep retrying this case.

Reply:
In that case can't we just check for the result of the task after the kill task RPC fails? If it's present, we can abort the kill task.

Reply:
Sounds good!